packages/client/src/types.d.ts

```diff
@@ -67,9 +67,8 @@ declare type LeaderType = {
   score: number
 }
 
-declare type ItemActionType = {
-  payload: UserType
-  type: string
+declare type TAction = {
+  [k: string]: any
```

Comment:
Очень странный тип, `TAction` используется в одном месте сейчас, можно же сделать вот так в `packages/client/src/store/reducers/user`:
```ts
type UserDataAction = {
  type: userTypes.SET_USER_DATA,
  payload: UserType
}

type UserProfileAction = {
  type: userTypes.SET_USER_PROFILE,
  payload: ProfileParams
}

type UserAvatarAction = {
  type: userTypes.SET_USER_AVATAR,
  payload: string
}

type UserAction = UserDataAction | UserProfileAction | UserAvatarAction;

export function userReducer(
  state: UserState = initState,
  { type, payload }: UserAction
): UserState {
```

Тогда в редьюсере у вас `payload` всегда будет соответствовать типу события, но далее увидел такую проблему: 
![Снимок экрана 2023-02-27 в 10 53 25](https://user-images.githubusercontent.com/54863186/221505355-3e40d4b4-d49b-43e6-a810-ae55547a3bcc.png)
То есть у вас `user` – это `UserType`, но вы сохраняете туда `payload: ProfileParams`, хотя `UserType` и `ProfileParams` не идентичны, отличаются в наличие `id`, немного странно как-то. Если я правильно понимаю, то изменение профиля (событие `userTypes.SET_USER_PROFILE`) происходит тогда, когда пользователь известен (его `id`), это видно по следующей строчке:
```tsx
export const PrivateProfile = withAuth(Profile)
```
Кроме того, это логично: менять данные профиля можно только тогда, когда профиль пользователя и сам пользователь есть.
Тогда можно сделать так в редьюсере:
```ts
    case userTypes.SET_USER_PROFILE:
      return {
        user: state.user ? { ...state.user, ...payload, } : null
      }
```

Reply:
@Brarion будет ошибка ts если типизировать action таким образом поскольку у нас подключается thunk, и экшен может становится не только объектом, но и функцией. Ошибка будет диспатче типа dispatch(setUser(data)), например, где setUser - создатель асинхронного действия. Например, в Login.txs ts будет говорить что Argument of type '(dispatch: Dispatch) => Promise<void>' is not assignable to parameter of type 'ItemActionType'.  Попробуй, зайти в этот файл с твоими правками. Собственно, типы setUser и TAction решают эту проблему. Я обратился за помощью к наставнику, он предложил такое решение, по другому не знаю как
вот ссылка на проблему для понимания https://github.com/reduxjs/redux-thunk/issues/333

Reply:
@CyberDasein а вот так не будет работать?
https://github.com/22-team-007/game-crocodile/commit/f54e6fa424dbf410d825e2dca698164638c0d715
Извините, что запушил вам ветку новую, я потом удалю)

Reply:
@Brarion аж 3 файла теперь красным :(

![image](https://user-images.githubusercontent.com/52383546/221584804-31aa3650-5c3d-4503-8d27-3789c3b09b7d.png)

Reply:
А можешь скинуть скринами, что за ошибки везде?

Reply:
Странно, что у меня их нет-_-

Reply:
Скринами не оч удобно... 
у store - const store: any
'store' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer. 
у AppDispatch - type AppDispatch = any
Type alias 'AppDispatch' circularly references itself

Reply:
в Login.ts - const dispatch: any
'dispatch' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.

Reply:
А `yarn install` выполнял? Может, какие-то библиотеки странно ведут себя?

Reply:
а почему у меня тогда есть они?)

Reply:
Возможно, наоборот, у меня они есть)

Reply:
Давай крч попробуем `node_modules` удалить, снова `yarn install` сделать, а затем отпишемся тут, есть ли ошибки

Reply:
Точно да, yarn install помог

Reply:
Пожалуйста)
Бывает такое, если видишь что-то странное – удаляй `node_modules`)
И ещё, вы используете `createStore`, я тут увидел, что он депрекейтнут, если будет время и желание, лучше избавиться от этого
![Снимок экрана 2023-02-27 в 18 14 47](https://user-images.githubusercontent.com/54863186/221602477-8d72201c-0a88-41e4-9462-c22455899395.png)

Reply:
В целом, на проект не повлияет, но лучше депрекейтнутые штуки не трогать

Reply:
Окей. То есть использовать toolkit?

Reply:
Судя по описанию, да, но там написано ещё про такое решение:
```ts
import { legacy_createStore as createStore} from 'redux'
```
оно тоже не очень, но зато зачеркиваться функция не будет)
Отталкивайся от своего свободного времени и размеру беклога)